js/src/personal/riyuan/src/react/nav.module.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useRef } from 'react'
+import { Text, Input } from '@mantine/core'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import {
+  faMagnifyingGlass,
+  faHouse,
+  faPerson,
+  faGamepad,
+  faMap,
+  faPowerOff,
+  faSun,
+  faMoon,
+} from '@fortawesome/free-solid-svg-icons'
+import navStyles from '../css/nav.module.css'
+import profilePic from '../../assets/images/pfp.jpg'
+
+const Nav: React.FC = () => {
+  const shellRef = useRef<HTMLElement>(null)
+  const toggleRef = useRef<HTMLElement>(null)
+  const searchBtnRef = useRef<HTMLLIElement>(null)
+  const modeSwitchRef = useRef<HTMLDivElement>(null)
+  const modeTextRef = useRef<HTMLSpanElement>(null) // Correct type for ref
+
+  useEffect(() => {
+    const body = document.body
+    const shell = shellRef.current
+    const toggle = toggleRef.current
+    const searchBtn = searchBtnRef.current
+    const modeSwitch = modeSwitchRef.current
+    const modeText = modeTextRef.current
+
+    if (!shell || !toggle || !searchBtn || !modeSwitch || !modeText) {
+      console.error('One or more elements not found')
+      return
+    }
+
+    const handleToggleClick = () => {
+      shell.classList.toggle(navStyles.close)
+    }
+
+    const handleSearchClick = () => {
+      shell.classList.remove(navStyles.close)
+    }
+
+    const handleModeSwitchClick = () => {
+      body.classList.toggle(navStyles.dark)
+      const sunIcon = document.querySelector(`.${navStyles.sun}`) as HTMLElement
+      const moonIcon = document.querySelector(
+        `.${navStyles.moon}`,
+      ) as HTMLElement
+      if (body.classList.contains(navStyles.dark)) {
+        if (modeText) modeText.innerText = 'Day Mode'
+        if (sunIcon) sunIcon.style.display = 'block'
+        if (moonIcon) moonIcon.style.display = 'none'
+      } else {
+        if (modeText) modeText.innerText = 'Night Mode'
+        if (sunIcon) sunIcon.style.display = 'none'
+        if (moonIcon) moonIcon.style.display = 'block'
+      }
+    }
+
+    toggle.addEventListener('click', handleToggleClick)
+    searchBtn.addEventListener('click', handleSearchClick)
+    modeSwitch.addEventListener('click', handleModeSwitchClick)
+
+    // Cleanup event listeners on component unmount
+    return () => {
+      toggle.removeEventListener('click', handleToggleClick)
+      searchBtn.removeEventListener('click', handleSearchClick)
+      modeSwitch.removeEventListener('click', handleModeSwitchClick)
+    }
+  }, [])
+
+  return (
+    <div className={navStyles.nav}>
```

Comment:
This does not use Mantine at all. Don't just copy someone else's code. You don't have the experience to tell if what they're doing is good or bad.